From 99da6ce228cb0c627a049a00f7f6d7cbd58e494f Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Tue, 21 May 2024 09:34:44 +0800
Subject: [PATCH] RunControl: Add missing code migration to std::shared_ptr

In `IDevice: Replace QSharedPointer with std::shared_ptr' (#536681), the
type of IDeviceConstPtr has changed to use std::shared_ptr, but the code
in RunControl was not updated, leading to compilation error:

error: no member named 'isNull' in 'IDeviceConstPtr'
  421 |     if (!device.isNull() && device->type() == ...
      |          ~~~~~~ ^
1 error generated.

Change-Id: I812034f37b14c9c62040f09e039a68eb4edfd4d6
---
 src/plugins/projectexplorer/runcontrol.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/projectexplorer/runcontrol.cpp b/src/plugins/projectexplorer/runcontrol.cpp
index 6085a72083..02df4037af 100644
--- a/src/plugins/projectexplorer/runcontrol.cpp
+++ b/src/plugins/projectexplorer/runcontrol.cpp
@@ -418,7 +418,7 @@ void RunControl::setDevice(const IDevice::ConstPtr &device)
     QTC_CHECK(!d->device);
     d->device = device;
 #ifdef WITH_JOURNALD
-    if (!device.isNull() && device->type() == ProjectExplorer::Constants::DESKTOP_DEVICE_TYPE) {
+    if (device && device->type() == ProjectExplorer::Constants::DESKTOP_DEVICE_TYPE) {
         JournaldWatcher::instance()->subscribe(this, [this](const JournaldWatcher::LogEntry &entry) {
 
             if (entry.value("_MACHINE_ID") != JournaldWatcher::instance()->machineId())
-- 
2.45.1

